### DISCUSSION TITLE
WSL NixOS Setup Issue: Wrapped Shell Error

### DISCUSSION THREAD

**Discussion by miguelomp** [nix-community/NixOS-WSL#556] [Q&A]
Hello everyone,

I’m not sure if this is the right place to raise this issue, as I haven’t found any references regarding my case.

I just turned on my new Beelink N100, and the only thing I did was install WSL and try to set up NixOS. When I run `wsl -d NixOS`, I receive the following error:

```Powershell
when trying to exec the wrapped shell

Caused by:
    No such file or directory (os error 2)
```

Does anyone have any idea what might be happening?

Thank you!

**Comment by nzbr** [comment]
Please try the latest tarball from main: https://github.com/nix-community/NixOS-WSL/actions/runs/11280670020/artifacts/2042148573
(The .tar.gz is inside the zip file)

Let me know if that fixes it for you

**Reply by laurierloi** [reply]
I ran into the same issue. Using the latest tarball suggested by @nzbr didn't solve the issue

EDIT: I also tried with [latest](https://github.com/nix-community/NixOS-WSL/actions/runs/11326664521/job/31496065035) with the same result

**Comment by zym787** [comment]
#516 There's a similar problem [here: wsl --version is not recognized means that you are running an old version of WSL2](https://github.com/nix-community/NixOS-WSL/issues/516#issuecomment-2244594271)

- So first look at the wsl version in PowerShell using` wsl --version` .If `wsl --version is not recognized`, an older version of WSL is used. Please visit [this website](https://github.com/microsoft/WSL/releases) to download the latest msi package

- Restart PowerShell after installation, and then use `wsl -d Nixos` to open Nixos
